# **Weather Analysis**
### A brief glance at Athenry's weather data as per metweb.ie.
## 1. Import data frame.

In [26]:
# Data frames
import pandas as pd

## 2. Collecting the data.

```bash
#! /bin/bash

 date
 echo "Downloading weather data"
 wget -O data/weather/`date +"%Y%m%d_%H%M%S.json"` https://prodapi.metweb.ie/observations/athenry/today
 echo "Weather data downloaded"
 date
```

Data is collected via metweb.ie's API and stored as JSON file in the following format YearMonthDay_HourMinuteSecond (e.g 20241127_220039, collected on Nov.27th 2024 at 22:00 39)

## 3. Anayzing the data.
### 3.1 Using Pandas to arrange and output the data for lean usability:

In [27]:
# Read the data.
df = pd.read_json('data/weather/20241127_220039.json')

### 3.2 Let's have a look at the head of the data (5 top lines):

In [28]:
# Show.
df.head()

name  temperature symbol weatherDescription              text  \
0  Athenry            0    15n         Fog / Mist            "Mist"   
1  Athenry            0    15n         Fog / Mist  "Fog thickening"   
2  Athenry           -1    15n         Fog / Mist             "Fog"   
3  Athenry           -2    15n         Fog / Mist  "Fog thickening"   
4  Athenry           -3    15n         Fog / Mist             "Fog"   

   windSpeed windGust cardinalWindDirection  windDirection  humidity  \
0          6        -                    NE             45        98   
1          6        -                    NE             45        99   
2          2        -                    NE             45        98   
3          6        -                    NW            315        98   
4          6        -                     N              0        97   

   rainfall  pressure    dayName       date reportTime  
0         0      1015  Wednesday 2024-11-27      00:00  
1         0      1015  Wednesday 2024-11-27      01:00  
2         0      1015  Wednesday 2024-11-27      02:00  
3         0      1015  Wednesday 2024-11-27      03:00  
4         0      1015  Wednesday 2024-11-27      04:00

### 3.3 Summary of descriptive statistics:

In [29]:
# Show.
df.describe()

temperature  windSpeed  windDirection  humidity  rainfall     pressure  \
count    22.000000  22.000000      22.000000   22.0000      22.0    22.000000   
mean     -1.363636   4.636364      98.181818   98.0000       0.0  1018.136364   
min      -3.000000   2.000000       0.000000   96.0000       0.0  1015.000000   
25%      -2.000000   2.000000       0.000000   97.2500       0.0  1015.250000   
50%      -1.000000   4.000000      67.500000   98.0000       0.0  1018.500000   
75%      -0.250000   6.000000     123.750000   99.0000       0.0  1020.000000   
max       0.000000   9.000000     315.000000   99.0000       0.0  1022.000000   
std       1.135801   2.300009     113.363569    0.9759       0.0     2.550359   

                      date  
count                   22  
mean   2024-11-27 00:00:00  
min    2024-11-27 00:00:00  
25%    2024-11-27 00:00:00  
50%    2024-11-27 00:00:00  
75%    2024-11-27 00:00:00  
max    2024-11-27 00:00:00  
std                    NaN

- count: Number of non-missing values.
- mean: Average value.
- std: Standard deviation.
- min: Minimum value.
- 25%: First quartile (25th percentile).
- 50%: Median (50th percentile).
- 75%: Third quartile (75th percentile).
- max: Maximum value.

### 3.4 Overview of the DataFrame's structure and contents:

In [30]:
print(df.info())


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 15 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   name                   22 non-null     object        
 1   temperature            22 non-null     int64         
 2   symbol                 22 non-null     object        
 3   weatherDescription     22 non-null     object        
 4   text                   22 non-null     object        
 5   windSpeed              22 non-null     int64         
 6   windGust               22 non-null     object        
 7   cardinalWindDirection  22 non-null     object        
 8   windDirection          22 non-null     int64         
 9   humidity               22 non-null     int64         
 10  rainfall               22 non-null     int64         
 11  pressure               22 non-null     int64         
 12  dayName                22 non-null     object        
 13  date   

We learn that the data set has 14 columns with 22 entries each which are data type object(8), int64(6) as well as datetime64(1)
Interesting is also the display of total memory usage of the DataFrame: 2.7+ KB.

In [ ]:
pd.set_option('display.width', 1000) # Printing column next to each other
print(df)

       name  temperature symbol weatherDescription              text  windSpeed windGust cardinalWindDirection  windDirection  humidity  rainfall  pressure    dayName       date reportTime
0   Athenry            0    15n         Fog / Mist            "Mist"          6        -                    NE             45        98         0      1015  Wednesday 2024-11-27      00:00
1   Athenry            0    15n         Fog / Mist  "Fog thickening"          6        -                    NE             45        99         0      1015  Wednesday 2024-11-27      01:00
2   Athenry           -1    15n         Fog / Mist             "Fog"          2        -                    NE             45        98         0      1015  Wednesday 2024-11-27      02:00
3   Athenry           -2    15n         Fog / Mist  "Fog thickening"          6        -                    NW            315        98         0      1015  Wednesday 2024-11-27      03:00
4   Athenry           -3    15n         Fog / Mist     

### 3.5 Summary
This file contains a collection of hourly weather observations from the synoptic station in Athenry, Co. Galway, for November 27th. The data, collected at 22:00 local time, provides 22 entries for each of the following variables: station name, dry bulb temperature (in whole degrees Celsius), weather description, windspeed (in knots), cardinal wind direction, relative humidity (percentage), rainfall (in millimeters), mean sea level pressure (in millibars), day of the week, date, and time of observation. November 27th was notably cold, with temperatures ranging from 0°C to below freezing. The day remained dry with no recorded rainfall but experienced fog to thick fog. Weather conditions were otherwise stable due to consistently high air pressure exceeding 1015 millibars throughout the day.

### End.